dev/ProgressRing/ProgressRing.cpp

```diff
@@ -84,29 +121,70 @@ void ProgressRing::OnBackgroundPropertyChanged(const winrt::DependencyObject&, c
 
 void ProgressRing::OnBackgroundColorPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
 {
-    if (auto&& player = m_player.get())
+    if (auto&& determinatePlayer = m_determinatePlayer.get())
+    {
+        if (auto const progressRingDeterminate = determinatePlayer.Source())
+        {
+            SetLottieBackgroundColor(progressRingDeterminate);
+        }
+    }
+
+    if (auto&& indeterminatePlayer = m_indeterminatePlayer.get())
     {
-        if (auto const progressRingIndeterminate = player.Source().try_as<AnimatedVisuals::ProgressRingIndeterminate>())
+        if (auto const progressRingIndeterminate = indeterminatePlayer.Source())
         {
             SetLottieBackgroundColor(progressRingIndeterminate);
         }
     }
 }
 
+void ProgressRing::OnOpacityPropertyChanged(const winrt::DependencyObject&, const winrt::DependencyProperty&)
```

Comment:
>OnOpacityPropertyChanged [](start = 19, length = 24)

Do you need to do this for the determinate one as well?

Reply:
Since the determinate animation never loops infinitely, the determinate player is stopped outside of rangebase updating, so doesn't need it.

Reply:
In that case you should not attach to the determinate player's opacity changed handler, since it isn't relevant for this

---
In reply to: [434280511](https://github.com/microsoft/microsoft-ui-xaml/pull/2584#discussion_r434280511) [](ancestors = 434280511)